recipes/eigen/all/conanfile.py

```diff
@@ -46,3 +48,5 @@ def package_info(self):
         self.cpp_info.components["eigen3"].includedirs = [os.path.join("include", "eigen3")]
         if self.settings.os == "Linux":
             self.cpp_info.components["eigen3"].system_libs = ["m"]
+        if self.options.MPL2_only:
+            self.cpp_info.defines = ["EIGEN_MPL2_ONLY"]
```

Comment:
```suggestion
            self.cpp_info.components["eigen3"].defines = ["EIGEN_MPL2_ONLY"]
```

Reply:
It's mandatory as we are using components here.

Reply:
Thanks!
I based my PR on an older version that didn't have the components. 
Why did we include components on a library that has only one component?